lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2443,30 +2471,32 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
                       diag::differentiable_attr_wrt_self_must_be_first);
           return;
         }
-        // The interface type of `self` must be used to construct the return
-        // type, but `isAnyClassReferenceType` cannot be called on it.
-        auto selfTy = original->getParent()->getSelfTypeInContext();
-        auto selfInterfaceTy = original->getParent()->getSelfInterfaceType();
-        if (selfTy->isAnyClassReferenceType() || selfTy->isExistentialType()) {
-          TC.diagnose(paramLoc,
-              diag::differentiable_attr_cannot_diff_wrt_objects_or_existentials,
-                      selfTy);
+        if (addWrtSelfRetTyOrDiagnose(TC, paramLoc, original, retElts))
           return;
-        }
-        retElts.push_back(selfInterfaceTy);
         break;
       }
       }
     }
-    // If collected `retElts` has only 1 element, use that element as adjoint's
-    // return type. Otherwise, make a tuple out of `retElts` as adjoint's return
-    // type.
-    assert(retElts.size() > 0 && "There should be at least one return type");
-    retTy = retElts.size() > 1
-      ? TupleType::get(retElts, ctx)
-      : retElts[0].getType();
   }
 
+  // This can happen when someone puts the attribute on an instance method with
+  // no paramters (other than the self parameter), and does not specify a wrt
+  // list.
+  if (retElts.size() == 0) {
+    TC.diagnose(attr->getLocation(), diag::differentiable_attr_wrt_nothing,
+                original->getName())
+      .highlight(original->getSourceRange());
```

Comment:
I will leave it as-is for now for local consistency, because other `TC.diagnose(...).highlight(...)`s in this func are indented 2 spaces.

A quick grepping around other code doesn't reveal a right answer. There are many places doing the exact same thing, and some use 2 spaces and others use 4.

Reply:
In the AD code base I initially used 2 but I have been switching to 4, since the predominant convention in the compiler uses 4. Please feel free to change them when you modify AD code paths.